v22.1/vault-db-secrets-tutorial.md

```diff
@@ -0,0 +1,440 @@
+        ~~~
+        ~~~txt
+        Keys
+        ----
+        5rPhNN6aG0dNRs97UmRYMmJH.iMXtW
+        ATKZIXMXMTcNQBkR4vwzvyd8.iMXtW
+        WHtZ4JKGLC8KYssMn5mCxKqU.iMXtW
+        lcAOenZ7s03BpPi8XKS0vsK3.iMXtW
+        yyufcaLu4eKWcnGzhes30t6M.iMXtW
+        ~~~
+
+    1. Now fetch the list of currently active SQL roles from the CockroachDB client:
+        {% include_cached copy-clipboard.html %}
+        ~~~shell
+        cockroach sql --url $CLI_DB_CONNECTION_URL --execute "show users;"
+        ~~~
+        ~~~txt
+                               username                       |                options                | member_of
+        ------------------------------------------------------+---------------------------------------+------------
+          admin                                               |                                       | {}
+          root                                                |                                       | {admin}
+          v-token-hc-crdb-rol-uqtqyca6neplilakhv55-1653513157 | VALID UNTIL=2022-05-25 22:12:42+00:00 | {}
+          v-token-hc-crdb-rol-0gzk3hwi7k7a9w1ggggq-1653515525 | VALID UNTIL=2022-05-25 22:52:10+00:00 | {}
+          v-token-hc-crdb-rol-varnevyjgjg9zgf62kps-1653515522 | VALID UNTIL=2022-05-25 22:52:07+00:00 | {}
+          v-token-hc-crdb-rol-wltbv25napuroomvzmok-1653515524 | VALID UNTIL=2022-05-25 22:52:09+00:00 | {}
+        ~~~
+
+1. Revoke credentials.
+    Try revoking a lease. Recall that a *lease* on a *role* in Vault terms corresponds to an actual credential pair on the CockroachDB cluster, i.e., a username/password combination for a SQL user.
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    vault lease revoke database/creds/crdb-role/XIpIBRM8FkQxD5B0ndB1lszY.iMXtW
+    ~~~
+    ~~~txt
+    All revocation operations queued successfully!
+    ~~~
+
+1. Provision a CockroachDB-client Vault policy.
+
+    As admin, provision a Vault policy for CockroachDB client operators to access the client credentials.
+
+    The purpose of the previous work is to make a dynamic secret that can be used to access the CockroachDB database by generating credentials on demand. Performing the above work (establishing the connection between the CockroachDB cluster and the Vault cluster, creating the template for database client credentials, etc.) required admin privileges. But for the work to be meaningful, a Vault user with more limited permissions must be able to access the generated credentials.
+
+    This policy will be used to access CockroachDB client credentials. In keeping with the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege), let's give it only the required ability to read the required credential. 
+
+    Vault policies are specified using HashiCorp Configuration Language (HCL). The following configuration specifies a policy of read access for the `crdb-role` credential.
```

Comment:
Probably optional, but you could link out to some information on HCL, maybe the spec https://github.com/hashicorp/hcl/blob/main/hclsyntax/spec.md the HashiCorp site links are more geared toward terraform: https://www.terraform.io/language/syntax/configuration but could also be informative